src/schema/schema_list.c

```diff
@@ -8,6 +8,62 @@
 
 #include "wt_internal.h"
 
+/*
+ * __wt_schema_get_tiered_uri --
+ *     Get the tiered handle for the named table.
+ */
+int
+__wt_schema_get_tiered_uri(WT_SESSION_IMPL *session, const char *uri, bool ok_incomplete,
+  uint32_t flags, WT_TIERED **tieredp)
+{
+    WT_DATA_HANDLE *saved_dhandle;
+    WT_DECL_RET;
+    WT_TIERED *tiered;
+    uint32_t i;
+
+    *tieredp = NULL;
+
+    saved_dhandle = session->dhandle;
```

Comment:
We also have a `WT_SAVE_DHANDLE()` macro that does essentially what you're doing here with `saved_dhandle`. I don't know if we have a stylistic preference for one over the other.

Reply:
Good catch. I forgot about that and definitely prefer to use it.